### ISSUE TITLE
@DateTimeFormat formatting using Spring ConversionService does not work inside Thymeleafs iteration 

### ISSUE THREAD

**Issue by olafos** [thymeleaf/thymeleaf-spring#37]
I'm struggling with the following issue:
Formatting fields annotated with @DateTimeFormat using Thymeleafs new ${{}} syntax doesn't work when the object containing formatted field is referenced inside th:each iteration.

However when i reference the iterated value directly using SpEL it's formatted correctly.

``` html
<tr data-th-each="record,iteration: ${page}">
 <td class="time" data-th-text="${{record.element.creationTime}}">Incorrect format (2013-11-04 17:17:53.0)</td>
 <td class="time" data-th-text="${{page.content[__${iteration.index}__].element.creationTime}}">Correct format (13-11-04)</td>
</tr>
```

In the above example the element is 

``` java
public class GroupHeader implements HasId<Long>, HasName, Selectable<Long>, Serializable {

    /**
     * 
     */
    private static final long serialVersionUID = 4657659227477994207L;

    @Getter
    private final Long id;

    @DateTimeFormat(style = "S-")
    @Getter
    @Setter
    private Date creationTime;

    @Getter
    private final String name;

    public GroupHeader(final Long id, final Date creationTime, final String name) {

        super();
        this.id = id;
        this.creationTime = creationTime;
        this.name = name;
    }

}
```

What might be the problem?

**Comment by danielfernandez** [comment]
That is actually the expected behaviour.

In order to being able to format a value using a @DateTimeFormat annotation, the object it references to must have been bound by Spring, and Spring must be able to recognize the expression being passed as referencing that bound object.

As an example, the _expression_ I'm referring to would be `obj.date` here:

``` html
<div>
   <p th:text="${{obj.date}}">...</p>
</div>
```

And also `obj.date` here (more common in forms):

``` html
<div th:object="${obj}">
   <p th:text="*{{date}}">...</p>
</div>
```

When Spring is being passed an expression like `obj.date`, it detects that the first component, `obj`, is actually a _bound_ object (normally an object used as a form model), and therefore applies not only the _general_ conversions (i.e. the converters and formatters registered at the `ConversionService`) but also the _binding-specific_ conversions (i.e. property editors and annotations), including the `@DateTimeFormat` annotation.

In your example, `page` is the bound object.

But in your iteration, the first part of the expression is not `page`, but `record`, which is not recognized by Spring as a bound object. In fact, it cannot be a bound object because it is an object you just created in the previous tag. So Spring is not able to recognize the expression and the annotations and property editors are not applied (however, the ConversionService would be, if it existed).

**Comment by olafos** [comment]
Thanks Daniel,

I understand now, it's a bit misleading nonetheless. So in my example, is
referencing "page" and using preprocessing to access the record the only
way to get the right formatting?

Cheers,

2013/11/5 Daniel Fernández notifications@github.com

**Comment by andidev** [comment]
I encountered the same problem. Filed an Issue #44 five days ago and now I found this issue.
I agree that it's a bit misleading. I'm also wondering is there a nicer way to do this without using preprocessing?

This is a quite common case in any application using tables with rows containing creation and modification dates. 
Having to use `data-th-text="${{page.content[__${iteration.index}__].element.creationTime}}"` instead of `data-th-text="${{record.element.creationTime}}"`is a big readability issue in my opinion.

**Comment by danielfernandez** [comment]
Note ticket #49 has been created in order to study future support for this scenario.

**Comment by danielfernandez** [comment]
Note this has been implemented and will be a part of Thymeleaf 2.1.3. See #49 for more information.
